### PR TITLE
make object prefix configurable

### PR DESCRIPTION
**Pull request by butonic**
## Description

Allow changing the object prefix. 
## Related Issue

Not all objectstorages will allow ':' in the object name, see https://github.com/minio/minio/issues/2954
## Motivation and Context

easier testing, objectstore incompatability
## How Has This Been Tested?

local minio instance, adding the param to the swift based test
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by butonic** [comment]
failing tests unrelated

**Comment by PVince81** [comment]
Makes sense :+1:

**Comment by butonic** [comment]
fixes https://github.com/owncloud/core/issues/17147

File: tests/objectstore/start-swift-ceph.sh [hunk 1]
```diff
@@ -85,6 +85,7 @@ cat > $thisFolder/swift.config.php <<DELIM
 		'username' => '$user',
 		'password' => '$pass',
 		'container' => 'owncloud-autotest$EXECUTOR_NUMBER',
+	    'objectPrefix' => 'autotest$EXECUTOR_NUMBER:oid:urn:',
```
Comment (PVince81) [inline tests/objectstore/start-swift-ceph.sh]: indent 😉

**Comment by PVince81** [comment]
:+1: makes sense

**Comment by PVince81** [comment]
huh, what am I reading

**Comment by PVince81** [comment]
ah ok, you just added the link... makes sense 😄

**Pull request merged**